.github/workflows/docs.yml

```
@@ -77,9 +77,15 @@ jobs:
       - name: Concretize
         run: spack -e . concretize -f
 
-        # Relies on cache-hit
+        # Relies on cache-hit(s)
+      - name: Install Dependencies
+        run: spack -e . install --only-concrete --show-log-on-error --only dependencies
+      
+        # Explicitly not using cache to get latest develop
       - name: Install Palace
-        run: spack -e . install --only-concrete --keep-stage --show-log-on-error
+        run: spack -e . install --only-concrete --show-log-on-error --only package --keep-stage --no-cache
+        # If you want to use the branch-local source instead (should we always do this?)
+        # spack -e . develop --path=$(pwd) local.palace@git."${{ github.head_ref || github.ref_name }}"=develop
```

```suggestion
        run: |
            spack -e . develop --path=$(pwd) local.palace@git."${{ github.head_ref || github.ref_name }}"=develop
            spack -e . install --only-concrete --show-log-on-error --only package --keep-stage --no-cache
```

This might just be best.